src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/SqlConnectionX.cs

```diff
@@ -53,25 +95,48 @@ public override string Database
         public override int ConnectionTimeout
             => throw new NotImplementedException();
 
-        /// <inheritdoc/>
+        /// <summary>
+        /// Gets or sets the connection string used to connect to the database.
+        /// </summary>
         [AllowNull]
         [DefaultValue("")]
         [RefreshProperties(RefreshProperties.All)]
         [SettingsBindable(true)]
         public override string ConnectionString
         {
-            get => throw new NotImplementedException();
-            set => throw new NotImplementedException();
+            get => _connectionString;
+            set
+            {
+                switch (State)
+                {
+                    case ConnectionState.Open:
+                    case ConnectionState.Connecting:
+                    case ConnectionState.Fetching:
+                    case ConnectionState.Executing:
+                        throw new InvalidOperationException("Can't change connection string on an active connection");
+                    default:
```

Comment:
This `default` branch doesn't do anything, you can delete it. Alternatively, you can rewrite the whole `switch` statement as `if (State is ConnectionState.Open or ConnectionState.Connecting or ...) throw ...`.